bench_autogptq/setup.sh

```diff
@@ -27,28 +30,82 @@ check_python() {
 }
 
 download_gptq_weights() {
-    # download the sample file if not exists
-    if [ ! -d "$GPTQ_WEIGHTS_FOLDER" ]; then
-        echo "Downloading GPT weights..."
-        huggingface-cli download TheBloke/Llama-2-7B-GPTQ --local-dir "$GPTQ_WEIGHTS_FOLDER" --exclude "*.git*" "*.md" "Notice" "LICENSE"
+    local MODEL_NAME="$1"
+
+    # Set download directory based on MODEL_NAME
+    if [ "$MODEL_NAME" = "llama" ]; then
+        DOWNLOAD_DIR="$LLAMA2_GPTQ_WEIGHTS_FOLDER"
+        MODEL_IDENTIFIER="TheBloke/Llama-2-7B-Chat-GPTQ"
+    elif [ "$MODEL_NAME" = "mistral" ]; then
+        DOWNLOAD_DIR="$MISTRAL_GPTQ_WEIGHTS_FOLDER"
+        MODEL_IDENTIFIER="TheBloke/Mistral-7B-Instruct-v0.1-GPTQ"
+    else
+        echo "Invalid MODEL_NAME. Supported values: 'llama', 'mistral'"
+        exit 1
+    fi
+
+    # Check if weights folder exists
+    echo "$DOWNLOAD_DIR"
+
+    if [ ! -d "$DOWNLOAD_DIR" ]; then
+        # Download weights using huggingface-cli
+        echo "Downloading weights to $DOWNLOAD_DIR..."
+        huggingface-cli download "$MODEL_IDENTIFIER" --local-dir "$DOWNLOAD_DIR" --exclude "*.git*" "*.md" "Notice" "LICENSE"
     else
-        echo "Weights already downloaded!"
+        echo "Weights already downloaded"
     fi
 }
 
+install_autogptq() {
+    if [ -d "$SCRIPT_DIR/AutoGPTQ" ]; then
+        echo "Removing existing AutoGPTQ directory..."
+        rm -rf "$SCRIPT_DIR"/AutoGPTQ
+    fi
+
+    git clone https://github.com/PanQiWei/AutoGPTQ.git "$SCRIPT_DIR"/AutoGPTQ
+    cd "$SCRIPT_DIR"/AutoGPTQ
+
+    # Now build
+
+    "$PYTHON_CMD" setup.py install
+
+    # come out of the dir
+    cd "$SCRIPT_DIR"
+}
+
 check_python
 
 if [ ! -d "$VENV_DIR" ]; then
     "$PYTHON_CMD" -m venv "$VENV_DIR"
     echo "Virtual environment '$VENV_DIR' created."
-fi
 
-# Activate the virtual environment
-# shellcheck disable=SC1091
-source "$VENV_DIR/bin/activate"
+    if [ -f "$VENV_DIR/bin/activate" ]; then
+        # shellcheck disable=SC1091
+        source "$VENV_DIR/bin/activate"
+    else
+        echo "Error: Unable to find virtual environment activation script."
+        exit 1
+    fi
+
+    "$PYTHON_CMD" -m pip install --upgrade pip > /dev/null
+    "$PYTHON_CMD" -m pip install -r "$SCRIPT_DIR/requirements.txt" --no-cache-dir > /dev/null
+
+    "$PYTHON_CMD" -m pip uninstall -y fsspec
+
+    # Install the required version of fsspec
+    "$PYTHON_CMD" -m pip install 'fsspec[http]>=2023.1.0,<=2024.2.0'
```

Comment:
why not moving this to the requirements.txt?

Reply:
Yeah, so even after I added this in requirements I am getting an error and the benchmarking process keeps getting broken. So I had to revert back to add this explicitly.